end-to-end-untargeted-metabolomics.Rmd

```diff
@@ -1,93 +1,744 @@
+adjusted data. 
+
+Use another ion as example...
+
 ## Correspondence
 
 - use *bad* settings (check maybe xcmsTutorials for examples)
 
+```{r}
+#' Default parameter for the grouping and apply them to the test ions BPC 
+param <- PeakDensityParam(sampleGroups = sampleData(data)$phenotype, bw = 30)
+plotChromPeakDensity(eic_cystine, param = param)
+plotChromPeakDensity(eic_Met, param = param)
+```
+
+- Good settings 
+```{r}
+#'Updating parameters
+param <- PeakDensityParam(sampleGroups = sampleData(data)$phenotype,
+                        minFraction = 1/3, binSize = 0.015, bw = 2)
+plotChromPeakDensity(eic_cystine, param = param)
+plotChromPeakDensity(eic_Met, param = param)
+```
+
+
+Now...
+
+```{r}
+#' Now apply to whole data
+data <- groupChromPeaks(data, param = param)
+```
+
+Test by checking an area with ions that are isomers of isoleucine and leucine 
+```{r}
+#' Extract chromatogram with signal for isomers leucine and isoleucine
+chr_test <- chromatogram(data, mz = 132.10191 + c(-0.01, 0.01),  
+                         rt = c(10, 180),aggregationFun = "max")
+plotChromPeakDensity(chr_test, simulate = FALSE)
+```
+Looks good 
+
+resulting obejct: 
+```{r}
+#' Definition of the features
+featureDefinitions(data) |> head()
+
+featureValues(data, method = "sum") |> head()
+```
+F00005 and F00006 contain missing values in some samples
+
+
 ## Gap filling
 
-- compare signal for filled-in and detected signal for QC samples. is it
-  correlated?
+Process to try to rescue some peaks that are only detected in a subset of samples. 
+```{r peaks found in a few QC not all, echo=FALSE}
+head(is.na(featureValues(data)))
+#' Extract the m/z-rt region for selected peaks
+mz_rt <- featureArea(data, features = c("FT00031", "FT00039"))
+mz_rt[, "rtmin"] <- mz_rt[, "rtmin"] - 2
+mz_rt[, "rtmax"] <- mz_rt[, "rtmax"] + 2
+
+#' Extract their EICs and plot them
+chrs <- chromatogram(data[c(1,4)], mz = mz_rt[, c("mzmin", "mzmax")],
+                     rt = mz_rt[, c("rtmin", "rtmax")])
+plot(chrs, col = c("red", "blue"), lwd = 2)
+
+
+```
+.. Discuss example 
+
+```{r}
+#' Number of missing values
+sum(is.na(featureValues(data)))
+```
+
+We can see quite a large amount of NA values in our data set. 
+
+```{r}
+data <- fillChromPeaks(data, param = ChromPeakAreaParam(), chunkSize = 5)
+
+#' How many missing values after
+sum(is.na(featureValues(data)))
+```
+With `fillChromPeaks` we could thus rescue signal for all but `r sum(is.na(featureValues(data)))` features. 
+
+Show the rescued peaks: 
+
+```{r echo=FALSE}
+#' Extract the m/z-rt region for selected peaks
+mz_rt <- featureArea(data, features = c("FT00031", "FT00039"))
+mz_rt[, "rtmin"] <- mz_rt[, "rtmin"] - 2
+mz_rt[, "rtmax"] <- mz_rt[, "rtmax"] + 2
+
+#' Extract their EICs and plot them
+chrs <- chromatogram(data[c(1,4)], mz = mz_rt[, c("mzmin", "mzmax")],
+                     rt = mz_rt[, c("rtmin", "rtmax")])
+plot(chrs, col = c("red", "blue"), lwd = 2)
+```
+
```

Comment:
maybe also add the comparison between filled and detected signals? to show that gap filling worked? it would be the last plot in section [gap filling](https://jorainer.github.io/xcmsTutorials/articles/xcms-preprocessing.html#gap-filling)

Reply:
added !